components/style/values/specified/length.rs

```diff
@@ -222,14 +222,128 @@ impl CharacterWidth {
     }
 }
 
+/// Same as Gecko
+const ABSOLUTE_LENGTH_MAX: f64 = (1 << 30) as f64;
+const ABSOLUTE_LENGTH_MIN: f64 = - (1 << 30) as f64;
+
+/// Helper to convert a floating point length to application units
+fn to_au_round(length: CSSFloat, au_per_unit: CSSFloat) -> Au {
+    Au(
+        ((length as f64) * (au_per_unit as f64))
+        .max(ABSOLUTE_LENGTH_MIN)
+        .min(ABSOLUTE_LENGTH_MAX)
+        .round() as i32
+    )
+}
+
+/// Represents an absolute length with its unit
+#[derive(Clone, PartialEq, Copy, Debug)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub enum AbsoluteLength {
+    /// An absolute length in application units
+    Au(Au),
```

Comment:
I don't think it makes a lot of sense to keep `AbsoluteLength::Au` around at all, does it?